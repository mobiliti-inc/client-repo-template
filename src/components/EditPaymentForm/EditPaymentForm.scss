@import "~variables";


.edit-payment-form {
    display: grid;
    margin-bottom: 50px;
}

.remove-card {
    margin-top: 18px;
    height: 21px;	
    width: 50%;	
    color: $fast-car-red;	
    font-family: $hanken-sans;	
    font-size: 18px;	
    font-weight: 500;	
    letter-spacing: -0.57px;	
    line-height: 21px;
    cursor: pointer;
}

.btn-container {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 10px;
    margin: 48px 0;
    height: 32px;	
    font-family: $futura;	
    font-size: 14px;	
    font-weight: bold;	
    line-height: 23px;	
    text-align: center;
    cursor: pointer;
    @media (max-width: $mobile-width) {
        grid-template-columns: 1fr;
        grid-row-gap: 10px;
        grid-column-gap: 0;
    }
}

.payment-card-holder {
    display: flex;
    align-items: center;
    cursor: pointer;
    height: 80px;
    margin-bottom: 20px;
}

.payment-card-icon {
    box-shadow: 0 2px 3px 0 rgba(0,0,0,0.2);
    left: 722px;
    top: 831px;
    margin: 0 28px 0 0;
    height: 50px;	
    width: 77px;
}

.payment-card-number {
    font-family: $futura;
    font-style: normal;
    color: $darker-blue;
    font-size: 20px;
    height: 31px;	
    width: 120px;	
    font-weight: bold;	
    letter-spacing: -0.44px;	
    line-height: 31px;
}

.state-code-wrapper {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 10px;
    @media (max-width: $mobile-width) {
        grid-template-columns: 1fr;
    }
}

.input-validation-message-label {
    @include input-validation-text;
    font-size: 15px;
    width: 100%;
 }