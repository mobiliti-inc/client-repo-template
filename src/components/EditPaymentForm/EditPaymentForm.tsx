import styles from "./EditPaymentForm.scss";
import TextInput, { INPUT_TYPES } from "../TextInput/TextInput";
import Button, { BUTTON_TYPES } from "../Button/Button";

type EditPaymentFormProps = {
	name: string,
	state: string,
	city: string,
	zipCode: string,
	isFormValid: boolean,
	handleInputChange: (...args: any[]) => any,
	updatePaymentMethod: (...args: any[]) => any,
	deletePaymentMethod: (...args: any[]) => any,
	imageUrl: string,
	lastFour: string,
	street: string,
	closeEditPaymentMethod: (...args: any[]) => any,
	isCardUpdating: boolean,
	error: string
};

const EditPaymentForm: React.SFC<EditPaymentFormProps> = props => {
	const {
		name,
		state,
		city,
		zipCode,
		imageUrl,
		lastFour,
		street,
		handleInputChange,
		updatePaymentMethod,
		deletePaymentMethod,
		isFormValid,
		closeEditPaymentMethod,
		isCardUpdating,
		error
	} = props;

	return (
		<div>
			<div styleName="payment-card-holder">
				{imageUrl && <img styleName="payment-card-icon" src={imageUrl} alt="payment card icon" />}
				<p styleName="payment-card-number">(......{lastFour})</p>
			</div>
			<div styleName="edit-payment-form">
				<TextInput
					placeholder="Name"
					label="Name"
					onChange={handleInputChange}
					type={INPUT_TYPES.TEXT}
					value={name}
					name="name"
					id="name"
					isValid
				/>
				<TextInput
					placeholder="Street"
					label="Street"
					onChange={handleInputChange}
					type={INPUT_TYPES.TEXT}
					value={street}
					name="street"
					id="street"
					isValid
				/>
				<TextInput
					placeholder="City"
					label="City"
					onChange={handleInputChange}
					type={INPUT_TYPES.TEXT}
					value={city}
					name="city"
					id="city"
					isValid
				/>
				<div styleName="state-code-wrapper">
					<TextInput
						placeholder="State"
						label="State"
						onChange={handleInputChange}
						type={INPUT_TYPES.TEXT}
						value={state}
						name="state"
						id="state"
						isValid
					/>

					<TextInput
						placeholder="Zip Code"
						label="Zip Code"
						onChange={handleInputChange}
						type={INPUT_TYPES.TEXT}
						value={zipCode}
						name="zipCode"
						id="zipCode"
						isValid
					/>
				</div>

				{error.length > 0 && <label styleName="input-validation-message-label">{error}</label>}

				<div role="presentation" onClick={deletePaymentMethod} styleName="remove-card">
					Remove Card
				</div>

				<div styleName="btn-container">
					<Button
						type={BUTTON_TYPES.STANDARD}
						onClick={updatePaymentMethod}
						disabled={!isFormValid}
						bordered
						showLoader={isCardUpdating}
					>
						Save Changes
					</Button>

					<Button type={BUTTON_TYPES.STANDARD} onClick={closeEditPaymentMethod} bordered>
						Close
					</Button>
				</div>
			</div>
		</div>
	);
};

EditPaymentForm.defaultProps = {};

export default CSSModules(EditPaymentForm, styles, { allowMultiple: true });
