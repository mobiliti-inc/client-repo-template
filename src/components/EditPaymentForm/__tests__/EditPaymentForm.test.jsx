import EditPaymentForm from '../EditPaymentForm';

const defaultProps = {
	name: 'fake text',
	state: 'fake text',
	city: 'fake text',
	zipCode: 'fake text',
	isFormValid: true,
	handleInputChange: jest.fn(),
	updatePaymentMethod: jest.fn(),
	deletePaymentMethod: jest.fn(),
	imageUrl: 'fake text',
	lastFour: 'fake text',
	street: 'fake text',
	closeEditPaymentMethod: jest.fn(),
	isCardUpdating: false,
	error: 'fake error',
};

describe('EditPaymentForm component', () => {
	it('should render', () => {
		const tree = shallow(<EditPaymentForm {...defaultProps} />);
		matchSnapshot(tree);
	});
});
