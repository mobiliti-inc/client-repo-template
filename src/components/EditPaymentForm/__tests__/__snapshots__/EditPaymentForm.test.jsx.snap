// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPaymentForm component should render 1`] = `
<div>
  <div
    className="payment-card-holder"
  >
    <img
      alt="payment card icon"
      className="payment-card-icon"
      src="fake text"
    />
    <p
      className="payment-card-number"
    >
      (......
      fake text
      )
    </p>
  </div>
  <div
    className="edit-payment-form"
  >
    <TextInput
      changePasswordVisibilityStatus={[Function]}
      className=""
      dark={false}
      disabled={false}
      errorMessage=""
      id="name"
      inputDidChange={false}
      isValid={true}
      label="Name"
      name="name"
      onBlur={null}
      onChange={[MockFunction]}
      placeholder="Name"
      shouldShowVisibility={false}
      type="text"
      value="fake text"
    />
    <TextInput
      changePasswordVisibilityStatus={[Function]}
      className=""
      dark={false}
      disabled={false}
      errorMessage=""
      id="street"
      inputDidChange={false}
      isValid={true}
      label="Street"
      name="street"
      onBlur={null}
      onChange={[MockFunction]}
      placeholder="Street"
      shouldShowVisibility={false}
      type="text"
      value="fake text"
    />
    <TextInput
      changePasswordVisibilityStatus={[Function]}
      className=""
      dark={false}
      disabled={false}
      errorMessage=""
      id="city"
      inputDidChange={false}
      isValid={true}
      label="City"
      name="city"
      onBlur={null}
      onChange={[MockFunction]}
      placeholder="City"
      shouldShowVisibility={false}
      type="text"
      value="fake text"
    />
    <div
      className="state-code-wrapper"
    >
      <TextInput
        changePasswordVisibilityStatus={[Function]}
        className=""
        dark={false}
        disabled={false}
        errorMessage=""
        id="state"
        inputDidChange={false}
        isValid={true}
        label="State"
        name="state"
        onBlur={null}
        onChange={[MockFunction]}
        placeholder="State"
        shouldShowVisibility={false}
        type="text"
        value="fake text"
      />
      <TextInput
        changePasswordVisibilityStatus={[Function]}
        className=""
        dark={false}
        disabled={false}
        errorMessage=""
        id="zipCode"
        inputDidChange={false}
        isValid={true}
        label="Zip Code"
        name="zipCode"
        onBlur={null}
        onChange={[MockFunction]}
        placeholder="Zip Code"
        shouldShowVisibility={false}
        type="text"
        value="fake text"
      />
    </div>
    <label
      className="input-validation-message-label"
    >
      fake error
    </label>
    <div
      className="remove-card"
      onClick={[MockFunction]}
      role="presentation"
    >
      Remove Card
    </div>
    <div
      className="btn-container"
    >
      <Button
        bordered={true}
        buttonType="standard"
        className=""
        customStyles=""
        disabled={false}
        icon={null}
        inverted={false}
        onClick={[MockFunction]}
        ripple={false}
        showLoader={false}
        type="standard"
      >
        Save Changes
      </Button>
      <Button
        bordered={true}
        buttonType="standard"
        className=""
        customStyles=""
        disabled={false}
        icon={null}
        inverted={false}
        onClick={[MockFunction]}
        ripple={false}
        showLoader={false}
        type="standard"
      >
        Close
      </Button>
    </div>
  </div>
</div>
`;
